bc_obps/reporting/models/activity_source_type_base_schema.py

```diff
@@ -0,0 +1,38 @@
+from registration.models import BaseModel
+from django.db import models
+from registration.models import ReportingActivity
+from reporting.models import SourceType, BaseSchema, Configuration
+from django.db.models.constraints import UniqueConstraint
+
+
+class ActivitySourceTypeBaseSchema(BaseModel):
+    """Intersection table for Activity-SourceType-BaseSchema"""
+
+    # No history needed, these elements are immutable
+    reporting_activity = models.ForeignKey(
+        ReportingActivity, on_delete=models.DO_NOTHING, related_name="activity_source_type_base_schemas"
+    )
+    source_type = models.ForeignKey(
+        SourceType, on_delete=models.DO_NOTHING, related_name="activity_source_type_base_schemas"
+    )
+    base_schema = models.ForeignKey(
+        BaseSchema, on_delete=models.DO_NOTHING, related_name="activity_source_type_base_schemas"
+    )
+    valid_from = models.ForeignKey(Configuration, on_delete=models.DO_NOTHING, related_name="+")
+    valid_to = models.ForeignKey(Configuration, on_delete=models.DO_NOTHING, related_name="+")
+
+    class Meta:
+        db_table_comment = "Intersection table that assigns a base_schema as valid for a period of time given an activity-sourceType pair"
+        db_table = 'erc"."activity_source_type_base_schema'
+        constraints = [
+            UniqueConstraint(
+                fields=[
+                    'reporting_activity',
+                    'source_type',
+                    'base_schema',
+                    'valid_from',
+                    'valid_to',
```

Comment:
Shouldn't the valid_from / valid_to overlap check go in the `configuration` table then? And this table does not need constraints?

Reply:
I think we'll need to use the save() method in this table, the exclusion constraint won't work here

Reply:
There should be an exclusion check for the dates in the `configuration` table, but we also need to make sure that we don't let those `activity_source_type_base_schema` records overlap on configuration IDs, otherwise we'll have ambiguous data

Overriding the `save()` method might be more readable, and it's not going to be a performance penalty since it'll only run during data migration